README.md

```diff
@@ -2,92 +2,707 @@
+    df = pd.read_csv(f)
+
+# Or read directly with pandas (if pandas supports the scheme)
+df = pd.read_csv('osdf:///ospool/uc-shared/public/data.csv')
+
+# Method 2: Using PelicanFileSystem directly
+from pelicanfs.core import PelicanFileSystem
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+with pelfs.open('/ospool/uc-shared/public/data.csv', 'r') as f:
+    df = pd.read_csv(f)
+```
+
+## Getting an FSMap
+
+Some systems prefer a key-value mapper interface rather than a URL. Use `PelicanMap` for this:
+
+```python
+from pelicanfs.core import PelicanFileSystem, PelicanMap
+
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+mapper = PelicanMap("/namespace/path/dataset.zarr", pelfs=pelfs)
+
+# Use with xarray
+import xarray as xr
+ds = xr.open_dataset(mapper, engine='zarr')
+```
+
+**Note:** Use `PelicanMap` instead of fsspec's `get_mapper()` for better compatibility with Pelican's architecture.
+
+## Monitoring and Debugging
+
+### Access Statistics
+
+PelicanFS tracks cache access statistics to help diagnose performance issues. For each namespace path, it keeps the last three cache access attempts:
+
+```python
+from pelicanfs.core import PelicanFileSystem
+
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+
+# Perform some operations
+pelfs.cat('/ospool/uc-shared/public/data.txt')
+pelfs.cat('/ospool/uc-shared/public/data.txt')  # Second access
+pelfs.cat('/ospool/uc-shared/public/data.txt')  # Third access
+
+# Get access statistics object
+stats = pelfs.get_access_data()
+
+# Get responses for a specific path
+responses, has_data = stats.get_responses('/ospool/uc-shared/public/data.txt')
+
+if has_data:
+    for resp in responses:
+        print(resp)
+
+# Print all statistics in a readable format
+stats.print()
+```
+
+**Example output:**
+
+```
+{NamespacePath: https://cache1.example.com/ospool/uc-shared/public/data.txt, Success: True}
+{NamespacePath: https://cache1.example.com/ospool/uc-shared/public/data.txt, Success: True}
+{NamespacePath: https://cache2.example.com/ospool/uc-shared/public/data.txt, Success: False, Error: <class 'aiohttp.client_exceptions.ClientConnectorError'>}
+/ospool/uc-shared/public/data.txt: {NamespacePath: https://cache1.example.com/ospool/uc-shared/public/data.txt, Success: True} {NamespacePath: https://cache1.example.com/ospool/uc-shared/public/data.txt, Success: True} {NamespacePath: https://cache2.example.com/ospool/uc-shared/public/data.txt, Success: False, Error: <class 'aiohttp.client_exceptions.ClientConnectorError'>}
+```
+
+**What the statistics show:**
+- **NamespacePath**: The full cache URL that was accessed
+- **Success**: Whether the cache access succeeded (`True`) or failed (`False`)
+- **Error**: The exception type if the access failed (only shown on failures)
+
+This helps identify:
+- Which caches are being used for your requests
+- Cache reliability and failure patterns
+- Whether cache fallback is working correctly
```

Comment:
I think describing what these are should happen before describing how to access them via PelicanFS. Maybe move up into the top-level "Access Statistics" section?